scylla-cdc/src/stream_generations.rs

```diff
@@ -0,0 +1,407 @@
+                    "INSERT INTO {}(time, range_end, streams) VALUES (?, -1, {{{}, {}}});",
+                    TEST_STREAM_TABLE, TEST_STREAM_1, TEST_STREAM_2
+                ),
+                (Timestamp(chrono::Duration::milliseconds(
+                    GENERATION_1_MILLISECONDS,
+                )),),
+            )
+            .await
+            .unwrap();
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (key, time, expired) VALUES ('timestamps', ?, NULL);",
+                    TEST_GENERATION_TABLE
+                ),
+                (Timestamp(chrono::Duration::milliseconds(
+                    GENERATION_1_MILLISECONDS,
+                )),),
+            )
+            .await
+            .unwrap();
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (key, time, expired) VALUES ('timestamps', ?, NULL);",
+                    TEST_GENERATION_TABLE
+                ),
+                (Timestamp(chrono::Duration::milliseconds(
+                    GENERATION_2_MILLISECONDS,
+                )),),
+            )
+            .await
+            .unwrap();
+    }
+
+    // Create setup for tests.
+    async fn setup() -> Result<GenerationFetcher, Box<dyn Error>> {
+        let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+
+        let session = SessionBuilder::new().known_node(uri).build().await?;
+
+        create_test_db(&session).await;
+        populate_test_db(&session).await;
+
+        let generation_fetcher = GenerationFetcher::test_new(&Arc::new(session));
+
+        Ok(generation_fetcher)
+    }
+
+    #[tokio::test]
+    async fn test_fetch_all_generations() {
+        let fetcher = setup().await.unwrap();
+
+        let gen = fetcher.fetch_all_generations().await.unwrap();
+
+        assert_eq!(gen.len(), 2);
+
+        let test_gen = Generation(chrono::Duration::milliseconds(GENERATION_1_MILLISECONDS));
+        assert_eq!(gen[0], test_gen);
+
+        let test_gen = Generation(chrono::Duration::milliseconds(GENERATION_2_MILLISECONDS));
+        assert_eq!(gen[1], test_gen);
+    }
+
+    fn construct_uuid(s: &str) -> Uuid {
+        Uuid::parse_str(s).unwrap()
+    }
+
+    #[tokio::test]
+    async fn test_get_generation_by_timeuuid() {
+        let fetcher = setup().await.unwrap();
+
+        let gen = fetcher
+            .fetch_generation_by_timeuuid(&construct_uuid(TEST_UUID))
+            .await
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(
+            gen,
+            Generation(chrono::Duration::milliseconds(GENERATION_1_MILLISECONDS))
+        );
+    }
+
+    #[tokio::test]
+    async fn test_get_next_generation() {
+        let fetcher = setup().await.unwrap();
+
+        let gen = fetcher.fetch_all_generations().await.unwrap();
+
+        let gen1 = fetcher.fetch_next_generation(&gen[0]).await.unwrap();
+        assert!(gen1.is_none());
```

Comment:
It looks like the assertions are swapped in this test - the first invocation of `fetch_next_generation` should return the second generation, while second invocation of `fetch_next_generation` should return `None` because there is no more generations.